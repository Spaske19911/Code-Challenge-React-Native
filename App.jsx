import { StyleSheet, Text, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import HomeScreen from './src/screens/HomeScreen';
import DetailsScreen from './src/screens/DetailsScreen';
import {useEffect, useState} from "react";
import {getToDoList} from "./api/toDoList";


export default function App() {
  const Stack = createNativeStackNavigator();

  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Details" component={DetailsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
