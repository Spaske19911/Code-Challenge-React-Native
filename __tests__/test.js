import React from 'react';
import { act } from 'react-test-renderer';
import App from '../App';

import renderer from 'react-test-renderer';

test('renders home screen', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />).toJSON();
  });
  expect(tree).toMatchSnapshot();
});

test('renders loader when fetching data', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />).toJSON();
  });
  expect(tree).toMatchSnapshot();
});

test('renders data list', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />).toJSON();
  });
  expect(tree).toMatchSnapshot();
});

test('navigates to detail screen when an item is clicked', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />);
  });
  const instance = tree.root;
  const dataList = instance.findByProps({ 'data-testid': 'data-list' });
  await act(async () => {
    dataList.props.onPress(dataList.props.data[0]);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders back button on detail screen', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />);
  });
  const instance = tree.root;
  const dataList = instance.findByProps({ 'data-testid': 'data-list' });
  await act(async () => {
    dataList.props.onPress(dataList.props.data[0]);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

test('navigates back to home screen when back button is clicked', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />);
  });
  const instance = tree.root;
  const dataList = instance.findByProps({ 'data-testid': 'data-list' });
  await act(async () => {
    dataList.props.onPress(dataList.props.data[0]);
  });
  const detailScreen = instance.findByProps({ 'data-testid': 'detail-screen' });
  const backButton = instance.findByProps({ 'data-testid': 'back-button' });
  await act(async () => {
    backButton.props.onPress();
  });
  expect(tree.toJSON()).toMatchSnapshot();
});

test('pagination limits number of items shown', async () => {
  let tree;
  await act(async () => {
    tree = renderer.create(<App />);
  });
  const instance = tree.root;
  const dataList = instance.findByProps({ 'data-testid': 'data-list' });
  await act(async () => {
    fireEvent.scroll(dataList, { nativeEvent: { contentOffset: { y: 1000 } } });
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
