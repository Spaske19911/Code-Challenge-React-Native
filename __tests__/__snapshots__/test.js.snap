// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders data list 1`] = `null`;

exports[`renders home screen 1`] = `null`;

exports[`renders loader when fetching data 1`] = `null`;
